sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/SQLQuerySuite.scala

```diff
@@ -2384,4 +2384,12 @@ class SQLQuerySuite extends QueryTest with SQLTestUtils with TestHiveSingleton {
       }
     }
   }
+
+  test("SPARK-27814: test cast operation for partition key") {
+    withTable("t1") {
+      spark.sql("CREATE TABLE t1(c1 Int, c2 String) PARTITIONED BY (c3 Int)")
+      spark.sql("SELECT * FROM t1 WHERE CAST(c3 as String) = '0'").show()
```

Comment:
let's not use `show` in test cases, use `checkAnswer`

Reply:
done